tuna/miopen/driver/base.py

```diff
@@ -180,57 +226,71 @@ def decompose_input_t(self, db_obj):
 
     return True
 
-  def get_weight_t_id(self):
+  def get_weight_t_id(self) -> int:
     """Build 1 row in tensor table based on layout from fds param
      Details are mapped in metadata LAYOUT"""
-    w_dict = self.compose_weight_t()
+    ret_id: int = -1
+    w_dict: dict = {}
 
-    return self.insert_tensor(w_dict)
+    w_dict = self.compose_weight_t()
+    ret_id = self.insert_tensor(w_dict)
+    return ret_id
 
-  def parse_driver_line(self, line):
+  def parse_driver_line(self, line: str):
     """Parse line and set attributes"""
-    line = parse_line(line)
-    tok = line.split()
+
+    tok: list
+    tmp_line: str = parse_line(line)
+
+    tok = tmp_line.split()
     #pylint: disable=attribute-defined-outside-init
     self.cmd = tok[1]
     assert tok[1] != ''
 
     self.compose_fds(tok, line)
 
-  def compose_fds(self, tok, line):
+  def compose_fds(self, tok: list, line: str) -> bool:
     """Compose fds from driver line"""
-
+    tok1: str
+    tok2: str
+    f_digi_v: Union[int, str]
     for (tok1, tok2) in zip(tok[2::2], tok[3::2]):
       # the following would not work for a full name argument
       if tok1[0] == '-':
-        tok1 = tok1[1:]
-      if self.test_skip_arg(tok1):
+        flag_type: str = tok1[1:]
+      if self.test_skip_arg(flag_type):
         continue
-      tok2 = tok2.strip()
-      if tok2.isdigit():
-        tok2 = int(tok2)
-      tok1 = self.get_params(tok1)
-      if self.get_check_valid(tok1[0], tok2):
-        setattr(self, tok1[0], tok2)
+      flag_value: str = tok2.strip()
+      if flag_value.isdigit():
+        f_digi_v = int(flag_value)
+      else:
+        f_digi_v = flag_value
+
+      flag_sh_value = self.get_params(flag_type)
+      if self.get_check_valid(flag_sh_value[0], f_digi_v):
+        setattr(self, flag_sh_value[0], f_digi_v)
       else:
         raise ValueError(
             f'Invalid command line arg for {self.cmd}: {tok1[0]} - {tok2} line: {line}'
         )
-
     return True
 
-  def to_dict(self):
+  def to_dict(self) -> Dict[str, int]:
```

Comment:
Convolution object has string fields like layout and direction. Not strictly Dict[str, int]. Currently Dict[str, Union[str, int]]